src/coreclr/tools/aot/ILCompiler.ReadyToRun/ObjectWriter/MapFileBuilder.cs

```diff
@@ -411,5 +446,55 @@ private void WriteTitle(StreamWriter writer, string title)
             writer.WriteLine(title);
             writer.WriteLine(new string('-', title.Length));
         }
+
+        private IEnumerable<MethodInfo> EnumerateMethods()
+        {
+            DebugNameFormatter nameFormatter = new DebugNameFormatter();
+            TypeNameFormatter typeNameFormatter = new TypeString();
+            HashSet<MethodDesc> emittedMethods = new HashSet<MethodDesc>();
+            foreach (KeyValuePair<ISymbolDefinitionNode, MethodWithGCInfo> symbolMethodPair in _methodSymbolMap)
+            {
+                EcmaMethod ecmaMethod = symbolMethodPair.Value.Method.GetTypicalMethodDefinition() as EcmaMethod;
+                if (ecmaMethod != null && emittedMethods.Add(ecmaMethod))
+                {
+                    MethodInfo methodInfo = new MethodInfo();
+                    methodInfo.MethodToken = (uint)MetadataTokens.GetToken(ecmaMethod.Handle);
+                    methodInfo.AssemblyName = ecmaMethod.Module.Assembly.GetName().Name;
+                    methodInfo.Name = FormatMethodName(symbolMethodPair.Value.Method, typeNameFormatter);
+                    MapFileNode node = _nodeSymbolMap[symbolMethodPair.Key];
+                    Section section = _sections[node.SectionIndex];
+                    methodInfo.HotRVA = (uint)(section.RVAWhenPlaced + node.Offset);
+                    methodInfo.HotLength = (uint)node.Length;
+                    methodInfo.ColdRVA = 0;
+                    methodInfo.ColdLength = 0;
+                    yield return methodInfo;
+                }
+            }
+        }
+
+        private string FormatMethodName(MethodDesc method, TypeNameFormatter typeNameFormatter)
+        {
+            StringBuilder output = new StringBuilder();
+            if (!method.Signature.ReturnType.IsVoid)
+            {
+                output.Append(typeNameFormatter.FormatName(method.Signature.ReturnType));
+                output.Append(" ");
+            }
+            output.Append(typeNameFormatter.FormatName(method.OwningType));
+            output.Append("::");
+            output.Append(method.Name);
+            output.Append("(");
+            for (int paramIndex = 0; paramIndex < method.Signature.Length; paramIndex++)
+            {
+                if (paramIndex != 0)
+                {
+                    output.Append(", ");
+                }
+                output.Append(typeNameFormatter.FormatName(method.Signature[paramIndex]));
+            }
+            output.Append(")");
+            return output.ToString();
+        }
     }
 }
+;
```

Comment:
is this a leftover ';'?

Reply:
Weird, no idea how it got there, thanks for spotting, removed in updated commit.